File: GFG/Array/Day4/NonRepeatingElement.java
import java.util.*;

public class NonRepeatingElement {
     public int firstNonRepeating(int[] arr) {
        // code here
        HashMap<Integer,Integer> map=new HashMap<>();
        
        for(int n:arr){
            if(map.containsKey(n)){
                map.put(n,map.get(n)+1);
            }else{
                map.put(n,1);
            }
        }
        
        for(int n:arr){
            if(map.get(n)==1){
                return n;
            }
        }
        return 0;
    }
}

File: GFG/Array/Day4/NonRepeatingElement.py
class Solution:
    def firstNonRepeating(self, arr): 
        # code here
        map={}
        
        for num in arr:
            if num in map:
                map[num]=map[num]+1
            else:
                map[num]=1
                
        for num in arr:
            if(map[num]==1):
                return num
            
        return 0